Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -424,28 +424,33 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::substr)
         return js_string(vm, string);
 
     // FIXME: this should index a UTF-16 code_point view of the string.
-    auto string_length = (i32)string.length();
+    auto size = (i32)string.length();
 
-    auto start_argument = vm.argument(0).to_i32(global_object);
+    auto int_start = vm.argument(0).to_integer_or_infinity(global_object);
     if (vm.exception())
         return {};
+    if (Value(int_start).is_negative_infinity())
+        int_start = 0;
+    else if (int_start < 0)
+        int_start = max(size + (i32)int_start, 0);
 
-    auto start = start_argument < 0 ? (string_length - -start_argument) : start_argument;
+    auto length = vm.argument(1).value_or(js_undefined());
+    if (vm.exception())
+        return {};
```

Comment:
This can't throw, and `vm.argument()` OOB access already returns undefined

```suggestion
    auto length = vm.argument(1);
```

Reply:
:+1: